app/controllers/logins_controller.rb

```diff
@@ -2,38 +2,31 @@
 
 class LoginsController < ApplicationController
   layout 'devise'
+
   def new; end
 
   def otp
-    @mobile_number = params[:mobile_number]
-    user_exists = User.exists?(phone: @mobile_number)
-    service = LoginWithOtpService.instance
+    @mobile_number = login_params[:mobile_number]
+    user = User.find_by!(phone: @mobile_number)
```

Comment:
@deepakkumarnd To prevent random hits, we can implement rate limiting, for example, using Cloudflare or any other rate limiting service to restrict network traffic. So with this approach we need to hit db only once.